django/db/models/expressions.py

```diff
@@ -777,8 +777,10 @@ class When(Expression):
     def __init__(self, condition=None, then=None, **lookups):
         if lookups and condition is None:
             condition, lookups = Q(**lookups), None
-        if condition is None or not isinstance(condition, Q) or lookups:
-            raise TypeError("__init__() takes either a Q object or lookups as keyword arguments")
+        if isinstance(condition, Expression) and isinstance(condition.output_field, fields.BooleanField):
```

Comment:
The ORM code mostly does `hasattr(value, 'resolve_expression')` rather than `isinstance(value, Expression)`, shouldn't we keep that here for consistency?